docs/REST API/rest-api.md

```diff
@@ -8926,13 +8936,26 @@ Returns information about the specified software. By default, `versions` are sor
       "installer_id": 23,
       "team_id": 3,
       "uploaded_at": "2024-04-01T14:22:58Z",
+      "install": "automatic"
       "install_script": "sudo installer -pkg /temp/FalconSensor-6.44.pkg -target /",
       "pre_install_query": "SELECT 1 FROM macos_profiles WHERE uuid='c9f4f0d5-8426-4eb8-b61b-27c543c9d3db';",
       "post_install_script": "sudo /Applications/Falcon.app/Contents/Resources/falconctl license 0123456789ABCDEFGHIJKLMNOPQRSTUV-WX",
       "self_service": true,
+      "labels_exclude_any": [
+        {
+          "name": "Label name 1",
+          "id": 1
+        },
+        {
+          "name": "Label name 2",
+          "id": 2
+        }
+      ],
       "status": {
```

Comment:
### Dev note

When a any host enrolls to Fleet (macOS, Windows, Linux, iOS/iPadOS), we want to clear it's software `status` and cancel any pending software installs.

Also, when a host is deleted in Fleet (via the UI/API), we want to clear it's software `status` and cancel any pending software installs.

Reply:
@lucasmrod heads up, I added what we discussed during design review plus the "when a host is deleted"

Please let me know if you have any concerns with this.

cc @gillespi314

Reply:
FYI I closed the following bug since we're addressing it as part of this story: https://github.com/fleetdm/fleet/issues/21075